pkg/codegen/dotnet/gen_test.go

```diff
@@ -15,21 +13,27 @@ import (
 )
 
 func TestGeneratePackage(t *testing.T) {
-	test.TestSDKCodegen(t, "dotnet", GeneratePackage, typeCheckGeneratedPackage)
+	test.TestSDKCodegen(t, &test.TestSDKCodegenOptions{
+		Language:   "dotnet",
+		GenPackage: generatePackageWithVersion,
+		Checks: map[string]test.CodegenCheck{
+			"dotnet/compile": typeCheckGeneratedPackage,
+		},
+	})
+}
+
+// TODO replace this with GeneratePackage when https://github.com/pulumi/pulumi/pull/7938 lands.
+func generatePackageWithVersion(tool string, pkg *schema.Package, extraFiles map[string][]byte) (map[string][]byte, error) {
```

Comment:
It looks like either this version or the original code works fine, not sure which to prefer.